### PR TITLE
Remove dynamic naming for `DEV_MODE` polyfill support functions.

### PR DESCRIPTION
**Pull request by bicknellr**
These dynamic lookups cause their expression to have an implicit `any` type, which is disallowed internally.

### CODE REVIEW

File: packages/lit-element/src/lit-element.ts [hunk 1]
```diff
@@ -196,7 +197,7 @@ if (DEV_MODE) {
     };
     warnRemovedOrRenamed(this, 'render');
     warnRemovedOrRenamed(this, 'getStyles', true);
-    warnRemovedOrRenamed(this.prototype, 'adoptStyles');
+    warnRemovedOrRenamed(this.prototype as {}, 'adoptStyles');
```
Comment (bicknellr) [inline packages/lit-element/src/lit-element.ts:200]: This should be in a different PR.

Comment (bicknellr) [reply packages/lit-element/src/lit-element.ts]: Removed.

**Review by justinfagnani — APPROVED** [review]
LGTM, but there are non-timeout related IE11 errors. They don't look relates to this PR either, but I'm not sure when they might have crept in.

**Comment by bicknellr** [comment]
`scoped-registry-mixin > host element should apply static styles` has been flaky in main for a bit now. I went through some of our recent runs and added [a comment about this in #1150](https://github.com/lit/lit/issues/1550#issuecomment-926244492).
